### PR TITLE
fix: Stop downloading data on network change

### LINKED ISSUE #1765: Downloader progress bar issue
**Actual Behaviour**

There is an absurd behaviour in the progress bar on a certain followed pattern. The download progress bar refuses to close too.

**Expected Behaviour**

The progress bar should work fine.

**Steps to reproduce it**

1.Open the app.
2.Turn the internet connection off.
3.Turn it on again.
4. The download progress bar starts.
5. After it completes, turn the internet off again.
6. Turn it back on.

**Screenshots of the issue**

screenshot_2017-07-02-23-20-18-224_org fossasia openevent

**Would you like to work on the issue?**

Yes.

**Target Milestone to solve the issue**

[milestone 9](https://github.com/fossasia/open-event-android/milestone/32). @iamareebjamal please add milestone.

### PR DESCRIPTION
**Pull request by rohanagarwal94**
Fixes #1765 

Changes: 

- [x] Stops listening to network changes.
- [x] Recreation of progress bar object to prevent stacking of strings.
- [x] Create a new complete handler object every time to prevent doubling up of subscribing events.

@iamareebjamal @arp95 @Shailesh351 @shreyasnbhat @mahikaw please review.

### CODE REVIEW

File: android/app/src/main/java/org/fossasia/openevent/utils/NetworkUtils.java [hunk 1]
```diff
@@ -106,9 +104,9 @@ public static void checkConnection(WeakReference<Context> reference, final Netwo
                                 .observeOn(AndroidSchedulers.mainThread())
                                 .subscribe(isActive -> {
                                     if (isActive) {
-                                        listener.activeConnection();
+                                        OpenEventApp.postEventOnUIThread(new DataDownloadEvent());
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/utils/NetworkUtils.java]: Um, this isn't correct, we shouldn't be posting a download event

File: android/app/src/main/java/org/fossasia/openevent/utils/NetworkUtils.java [hunk 2]
```diff
@@ -106,9 +104,9 @@ public static void checkConnection(WeakReference<Context> reference, final Netwo
...
                                     } else {
-                                        listener.inactiveConnection();
+                                        Toast.makeText(reference.get(), R.string.waiting_for_network, Toast.LENGTH_LONG).show();
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/utils/NetworkUtils.java]: Same here

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

File: android/app/src/main/java/org/fossasia/openevent/utils/NetworkUtils.java (hunk 1 above)
Comment (rohanagarwal94) [reply android/app/src/main/java/org/fossasia/openevent/utils/NetworkUtils.java]: oh sorry, missed it. Updating it.

**Comment by rohanagarwal94** [comment]
@iamareebjamal updated.

**Comment by iamareebjamal** [comment]
As there are no tests, please manually check if it is working in all cases

**Comment by rohanagarwal94** [comment]
@iamareebjamal yeah did that. Swipe refreshes, Initial app start, app start from pause.

**Review by iamareebjamal — APPROVED** [review]

**Comment by rohanagarwal94** [comment]
@mariobehling @the-dagger please review.

**Comment by rohanagarwal94** [comment]
@Shailesh351 @arp95 @shreyasnbhat @mahikaw please review.

**Review by arp95 — APPROVED** [review]
LGTM.

**Pull request merged**